packages/kit/src/core/prerender/prerender.js

```diff
@@ -12,45 +12,31 @@ import { load_config } from '../config/index.js';
 import { get_route_segments } from '../../utils/routing.js';
 import { get_option } from '../../runtime/server/utils.js';
 
-/**
- * @typedef {import('types').PrerenderErrorHandler} PrerenderErrorHandler
- * @typedef {import('types').Logger} Logger
- */
-
 const [, , client_out_dir, results_path, verbose, env] = process.argv;
 
 prerender();
 
 /**
- * @param {Parameters<PrerenderErrorHandler>[0]} details
- * @param {import('types').ValidatedKitConfig} config
- */
-function format_error({ status, path, referrer, referenceType }, config) {
-	const message =
-		status === 404 && !path.startsWith(config.paths.base)
-			? `${path} does not begin with \`base\`, which is configured in \`paths.base\` and can be imported from \`$app/paths\` - see https://kit.svelte.dev/docs/configuration#paths for more info`
-			: path;
-
-	return `${status} ${message}${referrer ? ` (${referenceType} from ${referrer})` : ''}`;
-}
-
-/**
- * @param {Logger} log
- * @param {import('types').ValidatedKitConfig} config
- * @returns {PrerenderErrorHandler}
+ * @template T
+ * @param {import('types').Logger} log
+ * @param {'fail' | 'warn' | 'ignore' | ((details: T) => void)} input
+ * @param {(details: T) => string} format
+ * @returns {(details: T) => void}
  */
-function normalise_error_handler(log, config) {
-	switch (config.prerender.onError) {
-		case 'continue':
+function normalise_error_handler(log, input, format) {
+	switch (input) {
+		case 'fail':
 			return (details) => {
-				log.error(format_error(details, config));
+				throw new Error(format(details));
 			};
-		case 'fail':
+		case 'warn':
 			return (details) => {
-				throw new Error(format_error(details, config));
+				log.error(format(details));
```

Comment:
is there a log.warn or should there be one?

Reply:
there is; the only difference is it prints yellow rather than red. we're already using it for informational purposes when prerendering (to indicate redirects), so i think it's better to use `log.error` here to differentiate